sdk/resource/resource.go

```diff
@@ -17,95 +17,22 @@
 package resource
 
 import (
-	"sort"
-	"strings"
-
 	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/label"
 )
 
-// Resource describes an entity about which identifying information and metadata is exposed.
 type Resource struct {
-	sorted []core.KeyValue
-	keySet map[core.Key]struct{}
+	labels label.Set
 }
 
 // New creates a resource from a set of attributes.
 // If there are duplicates keys then the first value of the key is preserved.
```

Comment:
According to the spec, the Merge semantics are based more on resource precedence:

> The resulting resource MUST have all attributes that are on any of the two input resources. Conflicts (i.e. a key for which attributes exist on both the primary and secondary resource) MUST be handled as follows:
>
> If the value on the primary resource is an empty string, the result has the value of the secondary resource.
> Otherwise, the value of the primary resource is used.

I think if we did a similar thing to the existing `Merge` method and just built a slice with the primary labels being last and then sort/de-dupe that we would implement the spec.

Otherwise, I don't see any issue with the spec on changing the label precedence here. I do wonder if we have logic internal to this codebase that relies on this first-value-wins ordering. I would guess not, but it's just a guess.

Reply:
I've updated the comment to say that `New` uses last-value-wins, which I believe is an OTel-wide policy, and that `Merge` uses gives the first resource preference.

Reply:
> I do wonder if we have logic internal to this codebase that relies on this first-value-wins ordering.

I suspect the answer is no. I found a couple of tests that had to be changed, of course.